operator/api/v1alpha1/virtualippool_types.go

```diff
@@ -0,0 +1,144 @@
+	// Tolerations to deploy the Virtual IPs manager
+	// +optional
+	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+	// Spread constraints for the Virtual IPs
+	// NOTE: Not supported yet
+	// // +optional
+	// SpreadConstraints []SpreadConstraintSpec `json:"spreadConstraints,omitempty"`
+
+	// Virtual IP addresses to use
+	// +kubebuilder:validation:MinItems=1
+	Addresses []IPAddress `json:"addresses"`
+}
+
+// VirtualIPPoolStatus defines the observed state of VirtualIPPool
+type VirtualIPPoolStatus struct {
+	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// List of conditions for the VirtualIPPool
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	// +listType=map
+	// +listMapKey=type
+	Conditions []Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type" protobuf:"bytes,1,rep,name=conditions"`
+}
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+//+kubebuilder:resource:shortName=vipp
+
+// VirtualIPPool is the Schema for the virtualippools API
+type VirtualIPPool struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   VirtualIPPoolSpec   `json:"spec,omitempty"`
+	Status VirtualIPPoolStatus `json:"status,omitempty"`
+}
+
+// Compute the ConfigMap name for a pool
+func (v *VirtualIPPool) GetConfigMap() *corev1.ConfigMap {
+	return &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      v.GetName(),
+			Namespace: v.GetNamespace(),
+		},
+	}
+}
+
+// Compute the DaemonSet name for a pool
+func (v *VirtualIPPool) GetDaemonSet() *appsv1.DaemonSet {
+	return &appsv1.DaemonSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      v.GetName(),
+			Namespace: v.GetNamespace(),
+		},
+	}
+}
+
+// Set a condition on VirtualIPPool
+func (v *VirtualIPPool) SetCondition(kind string, status metav1.ConditionStatus, reason string, message string) {
+	setCondition(v.Generation, &v.Status.Conditions, kind, status, reason, message)
+}
+
+// Get a condition from VirtualIPPool
+func (v *VirtualIPPool) GetCondition(kind string) *Condition {
+	return getCondition(v.Status.Conditions, kind)
+}
+
+// Set Configured Condition
+func (v *VirtualIPPool) SetConfiguredCondition(status metav1.ConditionStatus, reason string, message string) {
+	v.SetCondition(configuredConditionName, status, reason, message)
+}
+
+// Set Available Condition
+func (v *VirtualIPPool) SetAvailableCondition(status metav1.ConditionStatus, reason string, message string) {
+	v.SetCondition(availableConditionName, status, reason, message)
+}
+
+// Set Ready Condition
+func (v *VirtualIPPool) SetReadyCondition(status metav1.ConditionStatus, reason string, message string) {
+	v.SetCondition(readyConditionName, status, reason, message)
+}
+
+// Get Ready Condition
+func (v *VirtualIPPool) GetReadyCondition() *Condition {
+	return v.GetCondition(readyConditionName)
+}
```

Comment:
just a minor note: randomly i saw this PR, not understand much but you can remove the comment, the function/method name are clean enough so self-documented

Reply:
Agree, but it's good practice in Golang to have a comment for every "exported" method (I'm not sure we do have a comment for all of them here... but ideally we should)

`golint` would complain if we do not have such comments (we should add `golint` validation in the CI btw)